packages/sources/eth-balance/src/endpoint/balance.ts

```diff
@@ -0,0 +1,60 @@
+import { Validator, Requester, AdapterError } from '@chainlink/ea-bootstrap'
+import { ExecuteWithConfig, InputParameters, AxiosResponse } from '@chainlink/types'
+import { Config } from '../config'
+
+export const supportedEndpoints = ['balance']
+
+export const inputParameters: InputParameters = {
+  addresses: ['addresses', 'result'],
+}
+
+interface AddressWithBalance {
+  address: string
+  balance: string
+}
+
+interface Address {
+  address: string
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const addresses = validator.validated.data.addresses || validator.validated.data.result
+
+  if (!addresses || !Array.isArray(addresses) || addresses.length === 0) {
```

Comment:
Because of our validation check on L21-22, address will never be undefined.

```suggestion
  if (!Array.isArray(addresses) || addresses.length === 0) {
```